CinemaTicketSalesSystem/CinemaTicketSalesSystem/App_Start/AutofacConfig.cs

```diff
@@ -18,13 +23,47 @@ public static IContainer ConfigureContainer()
             builder.RegisterControllers(typeof(MvcApplication).Assembly);
             builder.RegisterType<ApplicationDbContext>().AsSelf().InstancePerRequest();
             builder.Register<UserStore<ApplicationUser>>(c => new UserStore<ApplicationUser>(c.Resolve<ApplicationDbContext>())).AsImplementedInterfaces();
-            //builder.RegisterGeneric(typeof(UserStore<>)).As(typeof(IUserStore<>)).InstancePerRequest();
             builder.Register<IdentityFactoryOptions<ApplicationDbContext>>(c => new IdentityFactoryOptions<ApplicationDbContext>()
             {
                 DataProtectionProvider = new Microsoft.Owin.Security.DataProtection.DpapiDataProtectionProvider("CinemaTicketSalesSystem")
             }).InstancePerRequest();
             builder.RegisterType<ApplicationUserManager>().AsSelf().InstancePerRequest();
-            builder.RegisterType<DbService>().As<IDbService>().WithParameter("context", new ApplicationDbContext());
+            builder.RegisterType<MovieService>().As<IMovieService>().InstancePerRequest();
```

Comment:
there is very useful method in Autofac which can be used to register all services in ones:
 builder.RegisterAssemblyTypes(YourAssembly)
                .Where(t => t.Name.EndsWith("Service"))
                .AsImplementedInterfaces()
                .InstancePerRequest();